--- a/configure.ac
+++ b/configure.ac
@@ -122,14 +122,5 @@ AM_CONDITIONAL([ENABLE_ERROR_INJECTION], [test "x$enable_error_injection" = xyes])
 AM_CONDITIONAL([ENABLE_ALL_STATIC], [test "x$enable_all_static" = xyes])
 
-dnl Find pthread, if we have it. We do this first because we may set -pthread on CFLAGS
-dnl which can cause various macros to be defined (__REENTRANT on Darwin, for example)
-AX_PTHREAD([
-   AC_DEFINE([HAVE_PTHREAD], [1], [Define to enable POSIX threads support.])
-   LIBS="$PTHREAD_LIBS $LIBS"
-   CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
-   CC="$PTHREAD_CC"
-])
-
 AC_FUNC_ALLOCA
 
 AC_FIND_FUNC([isatty], [c], [#include <unistd.h>], [0])
