--- a/include/posixjmp.h
+++ b/include/posixjmp.h
@@ -25,7 +25,7 @@
 /* This *must* be included *after* config.h */
 
-#if defined (HAVE_POSIX_SIGSETJMP)
+#if defined (HAVE_POSIX_SIGSETJMP) && !defined (__wasi__)
 #  define procenv_t	sigjmp_buf
 
 #  define setjmp_nosigs(x)	sigsetjmp((x), 0)

--- a/lib/readline/posixjmp.h
+++ b/lib/readline/posixjmp.h
@@ -25,7 +25,7 @@
 /* This *must* be included *after* config.h */
 
-#if defined (HAVE_POSIX_SIGSETJMP)
+#if defined (HAVE_POSIX_SIGSETJMP) && !defined (__wasi__)
 #  define procenv_t	sigjmp_buf
 
 #  define setjmp_nosigs(x)	sigsetjmp((x), 0)

--- a/lib/readline/readline.c
+++ b/lib/readline/readline.c
@@ -590,7 +590,7 @@
       /* Save and restore _rl_top_level even though most of the time it
 	 doesn't matter. */
       memcpy ((void *)olevel, (void *)_rl_top_level, sizeof (procenv_t));
-#if defined (HAVE_POSIX_SIGSETJMP)
+#if defined (HAVE_POSIX_SIGSETJMP) && !defined (__wasi__)
       code = sigsetjmp (_rl_top_level, 0);
 #else
       code = setjmp (_rl_top_level);

--- a/lib/readline/callback.c
+++ b/lib/readline/callback.c
@@ -142,7 +142,7 @@ rl_callback_read_char (void)
   eof = 0;
 
   memcpy ((void *)olevel, (void *)_rl_top_level, sizeof (procenv_t));
-#if defined (HAVE_POSIX_SIGSETJMP)
+#if defined (HAVE_POSIX_SIGSETJMP) && !defined (__wasi__)
   jcode = sigsetjmp (_rl_top_level, 0);
 #else
   jcode = setjmp (_rl_top_level);
