--- a/gnu/getopt.c
+++ b/gnu/getopt.c
@@ -44,7 +44,10 @@
 # define _(msgid) gettext (msgid)
 /* When used standalone, flockfile and funlockfile might not be
    available.  */
-# if (!defined _POSIX_THREAD_SAFE_FUNCTIONS \
+# if defined __wasi__
+#  define flockfile(fp) ((void) (fp))
+#  define funlockfile(fp) ((void) (fp))
+# elif (!defined _POSIX_THREAD_SAFE_FUNCTIONS \
       || (defined _WIN32 && ! defined __CYGWIN__))
 #  define flockfile(fp) /* nop */
 #  define funlockfile(fp) /* nop */
