--- a/src/util.cc
+++ b/src/util.cc
@@ -976,6 +976,10 @@ double GetLoadAverage() {
 #elif defined(__HAIKU__)
 double GetLoadAverage() {
     return -0.0f;
 }
+#elif defined(__wasi__)
+double GetLoadAverage() {
+  return -0.0f;
+}
 #else
 double GetLoadAverage() {
   double loadavg[3] = { 0.0f, 0.0f, 0.0f };
--- a/src/util.cc
+++ b/src/util.cc
@@ -873,6 +873,8 @@ int GetProcessorCount() {
     &mask) == 0) {
     return CPU_COUNT(&mask);
   }
+#elif defined(__wasi__)
+  return 1;
 #elif defined(CPU_COUNT)
   cpu_set_t set;
   if (sched_getaffinity(getpid(), sizeof(set), &set) == 0) {
