--- a/src/timer/unix_timer.rs
+++ b/src/timer/unix_timer.rs
@@ -38,6 +38,7 @@ impl CPUTimer {
 }
 
 /// Read CPU execution times ('user' and 'system')
+#[cfg(not(target_os = "wasi"))]
 fn get_cpu_times() -> CPUTimes {
     use libc::{getrusage, rusage, RUSAGE_CHILDREN};
 
@@ -73,6 +74,15 @@ fn get_cpu_times() -> CPUTimes {
     }
 }
 
+#[cfg(target_os = "wasi")]
+fn get_cpu_times() -> CPUTimes {
+    CPUTimes {
+        user_usec: 0,
+        system_usec: 0,
+        memory_usage_byte: 0,
+    }
+}
+
 /// Compute the time intervals in between two `CPUTimes` snapshots
 fn cpu_time_interval(start: &CPUTimes, end: &CPUTimes) -> CPUInterval {
     CPUInterval {
--- a/src/benchmark/executor.rs
+++ b/src/benchmark/executor.rs
@@ -343,6 +343,9 @@ impl Executor for MockExecutor {
             ExitStatus::from_raw(0)
         };
 
+        #[cfg(target_os = "wasi")]
+        let status = ExitStatus::default();
+
         Ok((
             TimingResult {
                 time_real: Self::extract_time(command.get_command_line()),
