--- a/gettext-runtime/intl/gnulib-lib/getlocalename_l-unsafe.c
+++ b/gettext-runtime/intl/gnulib-lib/getlocalename_l-unsafe.c
@@ -659,6 +659,9 @@
       else
         /* It's the "C" or "POSIX" locale.  */
         return (struct string_with_storage) { "C", STORAGE_INDEFINITE };
 # endif
+#elif defined __wasi__
+      (void) category;
+      return (struct string_with_storage) { "C", STORAGE_INDEFINITE };
 #elif defined __ANDROID__
       /* Android API level >= 21 */
       struct __locale_t {
