--- a/gnu/argp-help.c
+++ b/gnu/argp-help.c
@@ -35,6 +35,11 @@
 # include <wchar.h>
 #endif
 
+#if defined __wasi__
+# define flockfile(stream) ((void) (stream))
+# define funlockfile(stream) ((void) (stream))
+#endif
+
 #ifdef _LIBC
 # include <libintl.h>
 # undef dgettext
--- a/lib/system.h
+++ b/lib/system.h
@@ -25,6 +25,19 @@
 #include <alloca.h>
 
 #include <sys/types.h>
+#include <stdint.h>
+
+#if defined __wasi__
+# ifndef major
+#  define major(device) ((unsigned int) (((uint64_t) (device) >> 32) & UINT32_MAX))
+# endif
+# ifndef minor
+#  define minor(device) ((unsigned int) ((uint64_t) (device) & UINT32_MAX))
+# endif
+# ifndef makedev
+#  define makedev(major_value, minor_value) ((dev_t) ((((uint64_t) (major_value)) << 32) | (uint32_t) (minor_value)))
+# endif
+#endif
 #include <ctype.h>
 
 /* IN_CTYPE_DOMAIN (C) is nonzero if the unsigned char C can safely be given
